### PR TITLE
(WIP) De-emphasise conda

### PR DESCRIPTION
**Pull request by RobinL**
See https://github.com/moj-analytical-services/splink/pull/2083#issuecomment-2010007832

The biggest change is 

> Developing Splink requires Python, as well as Poetry (the package manager we use to install Python package dependencies).
> Testing locally on the Spark, Athena, or PostgreSQL backends requires additional dependencies.
> 
> There are two ways to install these system dependencies: globally on your computer, or in an isolated conda environment.
> 
> These can be managed by either:
> 
> 1. Installing them manually, or
> 2. Using a `conda` environment
> 
> The decision of which approach to take is subjective.
> 
> Some developers will already be using Spark and will already have `poetry` installed, but are not familiar with `conda`. For these users, there's probably little advantage to using `conda`.
> 
> On the other hand, `conda` is particular suitable if:
> 
> - You're already a `conda` user, and/or
> - You're working in an environment where security policies prevent the installation of system level packages like Java
> - You don't want to do global
> 

You'll also see I've moved things around a bit to put the manual install instructions first .

I haven't checked the formatting too carefully.  Could you let me know if you're broadly happy with this, and i'll do a more thorough readthrough.

### CODE REVIEW

File: docs/dev_guides/changing_splink/development_quickstart.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Splink is a complex project with many dependencies.
 This page provides step-by-step instructions for getting set up to develop Splink.
 Once you have followed these instructions, you should be all set to start making changes.
 
-### Step 0: Unix-like operating system
+## Step 0: Unix-like operating system
```
Comment (zmbc) [inline docs/dev_guides/changing_splink/development_quickstart.md:7]: Curious on the thinking behind this -- putting this at the same level as the title seems strange. Perhaps the title should be an h1?

File: docs/dev_guides/changing_splink/development_quickstart.md [hunk 2]
```diff
@@ -15,50 +15,47 @@ instructions for how to do so.
...
 If you haven't already, make a [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) of the Splink repository.
 Clone **your fork** to whatever directory you want to work in with `git clone https://github.com/<YOUR_USERNAME>/splink.git`.
 
-### Automatic steps 2-3: Installation script
+## Step 2: Choose how to install system dependencies
 
-For convenience, we have created an automatic installation script that will install all dependencies for you.
-It will create an isolated [conda environment](https://conda.io/projects/conda/en/latest/user-guide/getting-started.html) called `splink`.
+Developing Splink requires Python, as well as Poetry (the package manager we use to install Python package dependencies).
+Testing locally on the Spark, Athena, or PostgreSQL backends requires additional dependencies.
```
Comment (zmbc) [inline docs/dev_guides/changing_splink/development_quickstart.md]: This isn't quite accurate to how I was using the word "locally." It does _not_ require any additional dependencies to test on PostgreSQL, for instance, if you run the Postgres server on a different machine.

My intention was that you can run the Spark and PostgreSQL tests completely locally (i.e., without having anything set up previously), but the Athena tests _require_ an AWS account to be set up. I think I'd prefer to reword this to make it more explicit:

```suggestion
Running Spark or PostgreSQL on your computer to test those backends requires additional dependencies.
Athena only runs in the AWS cloud, so to locally run the tests for that backend you will need to create an AWS account and
configure Splink to use it.
```

File: docs/dev_guides/changing_splink/development_quickstart.md [hunk 3]
```diff
@@ -15,50 +15,47 @@ instructions for how to do so.
...
+There are two ways to install these system dependencies: globally on your computer, or in an isolated conda environment.
 
-```sh
-./scripts/conda/development_setup_with_conda.sh
-```
+These can be managed by either:
 
-If you've run this successfully, you can skip to the "Step 4: Activating your environment(s)" section.
+1. Installing them manually, or
+2. Using a `conda` environment
```
Comment (zmbc) [inline docs/dev_guides/changing_splink/development_quickstart.md:39]: I think this section can be removed -- the manual vs automatic distinction is only relevant for conda, and is addressed below in the conda section.

File: docs/dev_guides/changing_splink/development_quickstart.md [hunk 4]
```diff
@@ -15,50 +15,47 @@ instructions for how to do so.
...
 
-If you would prefer to manually go through the steps to have a better understanding of what you are installing, continue
-to the next section.
+The decision of which approach to take is subjective.
 
-### Step 2: Non-Python-package dependencies
+Some developers will already be using Spark and will already have `poetry` installed, but are not familiar with `conda`. For these users, there's probably little advantage to using `conda`.
```
Comment (zmbc) [inline docs/dev_guides/changing_splink/development_quickstart.md]:
```suggestion
If you already have Python and Poetry installed (plus Java and PostgreSQL if you want to run the
Spark and PostgreSQL backends locally), there is probably little advantage to using `conda`.
```

I found the Spark part of this a little unclear. I also prefer the 2nd person here.

**Review by zmbc — APPROVED** [review]
Overall looks good, sorry for my Conda bias 😄

File: docs/dev_guides/changing_splink/development_quickstart.md (hunk 1 above)
Comment (RobinL) [reply docs/dev_guides/changing_splink/development_quickstart.md:7]: I was fiddling with the visual formatting to try and make it as clear as possible that there are two different options.   As you go down the header levels, the 'break'/split between headers becomes less and less pronounced

I've just looked, and it seems that the H1 level element comes from `mkdocs.yml`, specifically the H1 level [Development Quickstart](https://github.com/moj-analytical-services/splink/blob/afdd9acd8540f271ed3180679761d1a7a7f9fdc9/mkdocs.yml#L239) text in the `yml` turns into the H1 element.  

(As a second example, see e.g. [Building docs](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/build_docs_locally.html?h=building) vs [Building docs locally](https://github.com/moj-analytical-services/splink/blob/afdd9acd8540f271ed3180679761d1a7a7f9fdc9/docs/dev_guides/changing_splink/build_docs_locally.md#L1) and the rendered version [here](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/build_docs_locally.html?h=building)

image



I don't feel strongly about any of this but it means we can probably drop the `## Development quickstart` at the top of this guide because it's repeated
image.  

Feel free to choose whatever you prefer!

(I don't think we've understood/been consistent about this before, I've only just looked into it now)

File: docs/dev_guides/changing_splink/development_quickstart.md (hunk 3 above)
Comment (RobinL) [reply docs/dev_guides/changing_splink/development_quickstart.md:39]: Agreed!

Comment (zmbc) [reply docs/dev_guides/changing_splink/development_quickstart.md:39]: I don't have write access to this PR, so will make this change in mine.
